src/MiddlewareFactory.php

```diff
@@ -49,18 +49,13 @@
  * @psalm-type CallableType = callable(ServerRequestInterface, RequestHandlerInterface): ResponseInterface
  * @psalm-type MiddlewareParam = string|InterfaceType|CallableType|list<string|InterfaceType|CallableType>
  */
```

Comment:
can we mark this as `@final` in v3 and making it `final` in v4?
Not yet totally aware of soft BC breaks and if we are doing this.

cc @mezzio/technical-steering-committee

Reply:
I think that would be fine. Psalm/phpstan will flag it to the user, but the runtime will still allow extending it. It's almost like a deprecation notice for extension in a way.